### ISSUE TITLE
GNS3 VM cannot be imported in latest versions of VMware

### ISSUE THREAD

**Issue by AJNOURI** [GNS3/gns3-server#1184]
On **brand new windows installation + vmware installation,** when trying to open an imported GNS3 VM:

![selection_005_01_10](https://user-images.githubusercontent.com/1716020/31055824-1e05b320-a6c9-11e7-9ee1-852ff1bfdcf8.jpg)

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/1716020/31056028-e799f2c6-a6cb-11e7-8289-4b7089460f35.gif)

**Comment by grossmj** [comment]
What VMware version did you use?

**Comment by AJNOURI** [comment]
14.0.0 build-6661328

**Comment by grossmj** [comment]
Thanks, I suspect this problem is only with the newest version of VMware.

**Comment by AJNOURI** [comment]
Indeed Jeremy, it works fine with the prior release **12.5.7**

![image](https://user-images.githubusercontent.com/1716020/31056644-18848afa-a6d5-11e7-8bd7-47b99267cf29.png)

Maybe we should somehow inform users during GNS3 VM download.

**Comment by grossmj** [comment]
We need to figure this out. Let's keep that issue open for now ;)

**Comment by grossmj** [comment]
Seems this also happens with VMWare Fusion 10. Is this a temporary bug with the newest version of VMware or is this an issue with how we build the GNS3 VM.ova?

**Comment by noplay** [comment]
pciSlot number is annoying because we need that to make sure the OS see it
in correct order.

Le dim. 1 oct. 2017 à 22:21, Jeremy Grossmann <notifications@github.com> a
écrit :

**Comment by AJNOURI** [comment]
I have tested old GNS3 VMs it is the same behavior with VMWare 14.

**Comment by ziajka** [comment]
Updating thread. It seems to that VMWare has dropped the support for ExtraConfig or it's a bug on WMware side. We will try to contact with VMWare and get an answer.

**Comment by tani10** [comment]
I had the same problem.The solution: First you install GNS3 VM.ova on VMWare 12 and then just upgrade to VMWare 14. It works fine.

**Comment by grossmj** [comment]
I have posted on VMware forum about this problem: https://communities.vmware.com/message/2711239#2711239

**Comment by grossmj** [comment]
Note that VIRL has similar issues: https://learningnetwork.cisco.com/thread/120789 and https://learningnetwork.cisco.com/thread/120925

**Comment by jsicuran** [comment]
I did exactly what Tani10 mentioned   _"I had the same problem.The solution: First you install GNS3 VM.ova on VMWare 12 and then just upgrade to VMWare 14. It works fine."_      

No luck and it mucked up  my VIRL setup too. I opened ticket with VMware and posted link to one of the other threads here about this. This is nonsense to pay for an in place upgrade to 14 and it breaks your other tools.

**Comment by jsicuran** [comment]
Here is what I posted at VMware for I just purchased the 14 upgrade last week. This is just unacceptable. 
Plus I hear 12 will be EOL in Feb 18.

Note:I removed VMware key and it fixed it so now it runs but it blew up my Cisco VIRL and GNS 3 installations. This is unacceptable what kind of junk release is this. I rely on your product as a solid tool. I should not have to downgrade to get my other tools working because 14 has issues. See support thread at https://gns3.com/qa/unable-to-import-gns3-vm-2-0-3-i

**Comment by noplay** [comment]
We can strip network settings with that:
https://github.com/GNS3/gns3-vm/blob/master/fix_vmware_ova_network.py

But not sure why VMware dropped it. Could be intersting to see what is inside the file .ova when you export it in VMware 14

**Comment by ziajka** [comment]
Hi @noplay ,

Yes, stripping that config makes no issue (ESXi release is without ExtraConfig and it can be imported).

I tried to export in VMware 14, basically ExtraConfig is dropped at all. 

I'm thinking if we really need that ExtraConfig. We obtain IP in "a guest-way":

https://github.com/GNS3/gns3-server/blob/master/gns3server/controller/gns3vm/vmware_gns3_vm.py#L142

and only if it fails we fall-back to:

https://github.com/GNS3/gns3-server/blob/master/gns3server/controller/gns3vm/vmware_gns3_vm.py#L150

which requires correct order of interfaces. 

You know maybe if there are any case that we really need it?

We're also thinking to modify VMX file before running GNS3 VM (after import).

Thanks and cheers,
Dominik

**Comment by ziajka** [comment]
I think I was able to solve the issue, except hardware virt. enabled on default. Test VM will come with RC3 which should be in this week.

**Comment by tani10** [comment]
Sorry for that .It works fine to me
    On Monday, 16 October 2017, 05:13:33 EEST, AMILABS <notifications@github.com> wrote:  
 
 
I did exactly what Tani10 mentioned "I had the same problem.The solution: First you install GNS3 VM.ova on VMWare 12 and then just upgrade to VMWare 14. It works fine."

No luck and it mucked up my VIRL setup too. I opened ticket with VMware and posted link to one of the other threads here about this. This is nonsense to pay for an in place upgrade to 14 and it breaks your other tools.

**Comment by ziajka** [comment]
This version should run on VMware 14 and older:

https://github.com/GNS3/gns3-gui/releases/download/v2.1.0rc3/GNS3.VM.VMware.Workstation.2.1.0rc3.zip

Please test and if works fine then I will close this issue.

**Comment by bigbigmo** [comment]
IT WORK! THANKS!

**Comment by Gauravism2017** [comment]
where i can get v2..1.3 for windows to be installed ob local machine

**Comment by dwright726** [comment]
The matching local installation can be found here :)

https://github.com/GNS3/gns3-gui/releases/tag/v2.1.0rc3

**Comment by onpaws** [comment]
Would it possible to enable VT-x/EPT by default?
I just downloaded and installed  2.1.0rc3 image and see that it seems to be off by default.

![image](https://user-images.githubusercontent.com/182515/31842912-b108c474-b5f0-11e7-9e12-70f9a89aa67e.png)
![image](https://user-images.githubusercontent.com/182515/31842920-bd0e9d0c-b5f0-11e7-8e47-4856505391aa.png)

**Comment by onpaws** [comment]
It appears that appending 
`vhv.enable = "TRUE"`
to 
GNS3 VM.vmx 

enables VT-x/EPT

**Comment by AJNOURI** [comment]
works with rc3 VM

![selection_010_21_10](https://user-images.githubusercontent.com/1716020/31848310-f86ae542-b62f-11e7-9a3f-1cee4a6147ce.jpg)

**Comment by grossmj** [comment]
@AJNOURI do you have KVM support in the VM?

**Comment by AJNOURI** [comment]
Looks OK

![image](https://user-images.githubusercontent.com/1716020/31848526-15af9d0a-b635-11e7-8ed8-f258d65a46a2.png)

**Comment by AJNOURI** [comment]
By the way @grossmj 
Looks like Vmware 14 uses different default location for VMs (C:\Users\{user}\Documents\Virtual Machines)

and GNS3 looks for (C:\Users\{user}\Documents\Virtual Machines), that's why it cannot detect GNS3 VM not its IP.

I had to rename the default "Virtual Machines" to "My Virtual Machines" for GNS3 to detect GNS3 VM.

Should I open a new issue?

**Comment by grossmj** [comment]
Yes please.

**Comment by onpaws** [comment]
@AJNOURI 
You had to check the `Virtualize VT-x/EPT` checkbox yourself, right?
I'm suggesting it would be ideal if the appliance was already configured for that.

@grossmj 
If you could please direct me to the repo for the GNS3 VM I'd be happy to try and submit a PR.

**Comment by ziajka** [comment]
Hi @onpaws,

You're right, when you just load GNS3 VM (without using GNS3) then this option is not enabled. This is because VMware 14 doesn't allow to pass this value. I was trying different options and none of them were working. You may try to do this on your computer and see if you can export -> change VM -> import VM. Any ideas which may enabled it on default are very welcome.

In case you load VM via GNS3 then we add option "vhv.enable = "TRUE"" directly to VMX before start.
Cheers,
Dominik

**Comment by grossmj** [comment]
@onpaws this is the GNS3 VM repo, please feel free to try: https://github.com/GNS3/gns3-vm

> In case you load VM via GNS3 then we add option "vhv.enable = "TRUE"" directly to VMX before start.

Indeed, the only problem is when an user doesn't start the VM using GNS3, for example a remote GNS3 VM (ESXi etc.)

**Comment by onpaws** [comment]
Gotcha, it sounds like VMware appliance files don't support this particular setting.
Would be nice to know if that's deliberate or support just may be lagging on VMware's side.

Has anyone filed a bug with VMware before?

**Comment by grossmj** [comment]
I only posted on VMware community forums: https://communities.vmware.com/message/2714144#2714144

**Comment by AJNOURI** [comment]
@onpaws , indeed I had to check it. It was unchecked by default.

**Comment by grossmj** [comment]
VMware 14 ignores the vhv.enable parameter in the .ovf file inside the .ova. Can you confirm that the `Virtualize VT-x/EPT` is checked when starting the GNS3 VM using the GNS3 GUI?

**Comment by grossmj** [comment]
I have tried on Windows. Importing the last GNS3 VM 2.1 RC3 into VMware 12.x: `Virtualize VT-x/EPT` was automatically checked as expected. I did the same with VMware 14 and the option was not checked after importing the OVA. Starting the VM from the GNS3 GUI and the option was checked.

I think we cannot do anything more from our side. Maybe VMware is going to fix this or provide a way to automatically activate the `Virtualize VT-x/EPT`  setting on OVA import. Also, we are not the only ones affected by this problem: https://learningnetwork.cisco.com/thread/120789
